Library/ViewModels/PledgeViewModel.swift

```diff
@@ -573,3 +578,63 @@ private func attributedConfirmationString(with project: Project) -> NSAttributed
 
   return attributedString
 }
+
+// MARK: - Validation Functions
+
+private func amountValid(
+  project: Project,
+  pledgeAmountData: PledgeAmountData,
+  context: PledgeViewContext
+) -> Bool {
+  guard let backing = project.personalization.backing, context.isUpdating else {
+    return pledgeAmountData.isValid
+  }
+
+  return backing.pledgeAmount != pledgeAmountData.amount && pledgeAmountData.isValid
+}
+
+private func shippingRuleValid(
+  project: Project,
+  reward: Reward,
+  shippingRule: ShippingRule?,
+  context: PledgeViewContext
+) -> Bool {
+  if context.isCreating {
+    return !reward.shipping.enabled || shippingRule != nil
+  }
+
+  guard
+    let backing = project.personalization.backing,
+    let shippingRule = shippingRule,
+    context.isUpdating
+  else {
+    return false
+  }
+
+  return backing.locationId != shippingRule.location.id
+}
+
+private func paymentMethodValid(
+  project: Project,
+  paymentSourceId: String,
+  context: PledgeViewContext
+) -> Bool {
+  guard let backing = project.personalization.backing, context.isUpdating else {
+    return true
+  }
+
+  return backing.paymentSource?.id != paymentSourceId
+}
+
+private func allValuesChangedAndValid(
+  amountValid: Bool,
+  shippingRuleValid: Bool,
+  paymentSourceValid: Bool,
+  context: PledgeViewContext
+) -> Bool {
+  if context.isUpdating {
```

Comment:
In an updating context we enable the submit button if anything has changed and his valid. In a creating context we require all values to be valid.